### ISSUE TITLE
[Bug] TerraBlender, Terralilith, BetterNether/End Crash

### ISSUE THREAD

**Issue by greensad** [paulevsGitch/BetterNether#503]
### What happened?

After extensive testing I've confirmed that using TerraBlender, Terralilith, Better Nether or Better End in conjunction will crash the game when creating a new world, removing **any** one of these mods will allow the game to create a new world normally. BCLib without Better Nether or Better End will allow the game to create a new world normally. I've never encountered a crash like this before and so I'm not sure where the best place to post the issue would be so I'm opening an issue here and on TerraBlender's issue tracker as well: https://github.com/Glitchfiend/TerraBlender/issues/22

### BetterNether

6.0.9

### BCLib

1.2.5

### Fabric API

0.46.1

### Fabric Loader

0.12.12

### Minecraft

1.18.1

### Relevant log output

```shell
https://bytebin.lucko.me/VITCgd380c or https://crashy.net/pBp1LPn4rD8799dniMXP

**MultiMC Instance:** https://drive.google.com/file/d/1m5v6mpfDx3uHTJoUwXLfVbefaxp7x4IO/view?usp=sharing
```

### Other Mods

```shell
Better End 1.0.1
Terralilith 2.0.12
TerraBlender 1.1.0.70
Not Enough Crashes 4.1.4
```

**Comment by quiqueck** [comment]
Hm, from what I see this is due to cclic serialization of surface rules. Not sure why that is happening tho...

**Comment by quiqueck** [comment]
Hm, I think something is choking on our custom SurfaceRule Conditions when TerraBlender is present.

**Comment by TheButterbrotMan** [comment]
@quiqueck Is there any ETA for the fix? Probably could it take minutes, hours, days, weeks, years? :D ;)

**Comment by paulevsGitch** [comment]
Fix is already in code, so it will be in next release of BCLib

**Comment by quiqueck** [comment]
I am currently testing the fix, and BetterEnd may also need some updating.

**Comment by greensad** [comment]
Phenomenal work and initiative thus far to y'all!

**Comment by quiqueck** [comment]
The Fix for this crash should be in 6.0.10. Please re-open this issue if you still have the crash after upgrading. However, be aware that BetterNether and TerraBlender are still not compatible. We continue to track that progress in #504
